homeassistant/components/tfa_me/coordinator.py

```diff
@@ -0,0 +1,135 @@
+        self.entities_added = 0
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=self.poll_interval,
+            config_entry=self.config_entry,
+        )
+
+    async def _async_update_data(self):
+        """Request and update data."""
+        parsed_data = {}  # dict
+
+        # Try to get an IP for a mDNS host name:
+        # - when IP can be solved it returns the IP
+        # - when it is an IP it just returns the IP
+        if "-" in self.host:
+            # station ID, contains "-"
+            mdns_name = f"tfa-me-{self.host:}.local"
+            resolved_host = await self.resolve_mdns(mdns_name)
+        else:
+            resolved_host = self.host
+
+        # Try to update data from station URL: e.g. "http://192.168.1.38/sensors"
+        try:
+            # New TFA.me client
+            tfa_me_client = TFAmeClient(resolved_host, "sensors", log_level=1)
+
+            # Fetch all available sensors as JSON from TFA.me station
+            json_data = await tfa_me_client.async_get_sensors()
+
+            # New TFA.me data structure
+            tfa_me_data = TFAmeDataForHA(multiple_entities=True)
+
+            # Convert JSON to TFA.me data for HA
+            parsed_data = tfa_me_data.json_to_entities(json_data=json_data)
+            self.gateway_id = tfa_me_data.get_gateway_id()
+
+        # Specific error mapping
+        except (TFAmeTimeoutError, TFAmeConnectionError) as err:
+            # Device not reachable → after first start "NotReady", then "UpdateFailed"
+            msg: str = "Timeout general connection error: " + str(err.__doc__)
+            _LOGGER.error(msg)
+            if self.first_init == 0:
+                raise ConfigEntryNotReady(
+                    f"Cannot reach {resolved_host}: {err}"
+                ) from err
+            raise UpdateFailed(f"Connection problem: {err}") from err
+
+        except (TFAmeHTTPError, TFAmeJSONError) as err:
+            # Device responding but data invalid
+            msg: str = "HTTP or invalid response error: " + str(err.__doc__)
+            _LOGGER.error(msg)
+            raise UpdateFailed(f"Invalid response: {err}") from err
+
+        except TFAmeException as err:
+            # All other client errors
+            msg: str = "All other client errors: " + str(err.__doc__)
+            _LOGGER.error(msg)
+            if self.first_init == 0:
+                raise ConfigEntryNotReady(f"TFA.me client error: {err}") from err
+            raise UpdateFailed(f"TFA.me client error: {err}") from err
+
+        except Exception as err:
+            # Fallback for unexpected errors
+            msg: str = "Fallback for unexpected errors: " + str(err.__doc__)
+            _LOGGER.error(msg)
+            if self.first_init == 0:
+                raise ConfigEntryNotReady(f"Unexpected error: {err}") from err
+            raise UpdateFailed(f"Unexpected error: {err}") from err
```

Comment:
Some of these do the same thing, so why not:

(or similar)
```suggestion
        except (TFAmeHTTPError, TFAmeJSONError) as err:
            # Device responding but data invalid
            _LOGGER.exception("Invalid response")
            raise UpdateFailed(f"Invalid response: {err}") from err

        except (TFAmeTimeoutError, TFAmeConnectionError, TFAmeException, Exception) as err:
            _LOGGER.exception("Error occurred")
            if self.first_init == 0:
                raise ConfigEntryNotReady(
                    f"Cannot reach {resolved_host}: {err}"
                ) from err
            raise UpdateFailed(f"Connection problem: {err}") from err
```